### PR TITLE
Rephrase custom PBA file descriptions in configuration

### PR DESCRIPTION
**Pull request by shreyamalviya**
Related: #1020

### CODE REVIEW

**Review by VakarisZ — CHANGES_REQUESTED** [review]
First, let us resolve the PBA fix PR

File: monkey/infection_monkey/post_breach/actions/users_custom_pba.py [hunk 1]
```diff
@@ -31,37 +27,28 @@ class UsersPBA(PBA):
...
                     self.command = (DIR_CHANGE_WINDOWS % get_monkey_dir_path()) + WormConfiguration.custom_PBA_windows_cmd
-                    self.filename = WormConfiguration.PBA_windows_filename
-                else:
-                    file_path = os.path.join(get_monkey_dir_path(), WormConfiguration.PBA_windows_filename)
-                    self.command = DEFAULT_WINDOWS_COMMAND.format(file_path)
-                    self.filename = WormConfiguration.PBA_windows_filename
             elif WormConfiguration.custom_PBA_windows_cmd:
                 self.command = WormConfiguration.custom_PBA_windows_cmd
 
-    def _execute_default(self):
```
Comment (mssalvatore) [inline monkey/infection_monkey/post_breach/actions/users_custom_pba.py:61]: Can you explain this change? It think it's not ideal to be downloading files from the island in the constructor if we can avoid it. Also, it makes sense to me that downloading the file is part of execution; I wouldn't expect the initialization of the PBA to fail if there were a network outage or other condition that prevented the download.

**Review by mssalvatore — COMMENTED** [review]
Can you rework and include the tests from https://github.com/guardicore/monkey/pull/1020?

File: monkey/infection_monkey/post_breach/tests/actions/test_users_custom_pba.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
+import pytest
```
Comment (VakarisZ) [inline monkey/infection_monkey/post_breach/tests/actions/test_users_custom_pba.py:1]: Unit tests are nice, but not a silver bullet. There's an antipattern in TDD called the "mockery", when most of the functionality is mocked. I don't suggest refactoring the whole pba's to solve it, but I can see BB tests bringing more value here and still being trivial enough to implement (command execution at least). Other PBA's should be tested with BB's IMO.

File: monkey/monkey_island/cc/services/config_schema/monkey.py [hunk 3]
```diff
@@ -20,9 +20,9 @@
                     "title": "Linux post breach file",
                     "type": "string",
                     "format": "data-url",
-                    "description": "File to be executed after breaching. "
-                                   "If you want custom execution behavior, "
-                                   "specify it in 'Linux post breach command' field. "
+                    "description": "File to be uploaded after breaching. "
+                                   "If you want the file to be executed, "
+                                   "specify it in the 'Linux post breach command' field. "
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/services/config_schema/monkey.py:25]:
```suggestion
                    "description": "File will be uploaded after breaching. "
                                           "Use 'Linux post breach command' field to "
                                           "change permissions, run or delete the file."
```

File: monkey/monkey_island/cc/services/config_schema/monkey.py [hunk 4]
```diff
@@ -35,9 +35,9 @@
                     "title": "Windows post breach file",
                     "type": "string",
                     "format": "data-url",
-                    "description": "File to be executed after breaching. "
-                                   "If you want custom execution behavior, "
-                                   "specify it in 'Windows post breach command' field. "
+                    "description": "File to be uploaded after breaching. "
+                                   "If you want the file to be executed, "
+                                   "specify it in the 'Windows post breach command' field. "
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/services/config_schema/monkey.py:40]:
```suggestion
                    "description": "File will be uploaded after breaching. "
                                           "Use 'Windows post-breach command' field to "
                                           "run or delete the file."
```

File: monkey/monkey_island/cc/services/config_schema/monkey.py [hunk 5]
```diff
@@ -35,9 +35,9 @@
                     "title": "Windows post breach file",
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/services/config_schema/monkey.py:35]:
```suggestion
                    "title": "Windows post-breach file",
```

File: monkey/monkey_island/cc/services/config_schema/monkey.py [hunk 6]
```diff
@@ -20,9 +20,9 @@
                     "title": "Linux post breach file",
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/services/config_schema/monkey.py:20]:
```suggestion
                    "title": "Linux post-breach file",
```

**Review by VakarisZ — CHANGES_REQUESTED** [review]
We also need to change descriptions of PBA command fields. They should also contain examples. Linux description should be something like:
"Use this field to run custom commands or uploaded files on exploited machines. Example command `chmod +x ./my_script.sh; ./my_script.sh ; rm ./my_script.sh`"

File: monkey/monkey_island/cc/services/config_schema/monkey.py (hunk 3 above)
Comment (mssalvatore) [reply monkey/monkey_island/cc/services/config_schema/monkey.py:25]: "File will be uploaded" doesn't seem right as a description. "File to be uploaded" is what I would expect.

**Pull request merged**